tests/test_splitter.py

```diff
@@ -0,0 +1,631 @@
+                test_splits = [("query", 0.4 / 0.7), ("gallery", 0.3 / 0.7)]
+                splitter.MatchingReIDSplit(source, id_splits, test_splits)
+            with self.assertRaisesRegex(Exception, "Subset name"):
+                id_splits = [("train", 0.5), ("val", 0.2), ("test", 0.3)]
+                test_splits = [("_query", 0.4 / 0.7), ("gallery", 0.3 / 0.7)]
+                actual = splitter.MatchingReIDSplit(
+                    source, id_splits, test_splits
+                )
+                len(actual.get_subset_by_group("query"))
+
+        with self.subTest("wrong attribute name for person id"):
+            id_splits = [("train", 0.5), ("val", 0.2), ("test", 0.3)]
+            test_splits = [("query", 0.4 / 0.7), ("gallery", 0.3 / 0.7)]
+            actual = splitter.MatchingReIDSplit(source, id_splits, test_splits)
+            with self.assertRaisesRegex(Exception, "Unknown group"):
+                actual.get_subset_by_group("_gallery")
+
+    def _generate_detection_dataset(self, **kwargs):
+        append_bbox = kwargs.get("append_bbox")
+        with_attr = kwargs.get("with_attr", False)
+        nimages = kwargs.get("nimages", 10)
+
+        label_cat = LabelCategories()
+        for i in range(6):
+            label = "label%d" % (i + 1)
+            if with_attr is True:
+                attributes = {"attr0", "attr%d" % (i + 1)}
+            else:
+                attributes = {}
+            label_cat.add(label, attributes=attributes)
+        categories = {AnnotationType.label: label_cat}
+
+        iterable = []
+        attr_val = 0
+        totals = np.zeros(3)
+        objects = [(1, 5, 2), (3, 4, 1), (2, 3, 4), (1, 1, 1), (2, 4, 2)]
+        for img_id in range(nimages):
+            cnts = objects[img_id % len(objects)]
+            totals += cnts
+            annotations = []
+            for label_id, count in enumerate(cnts):
+                attributes = {}
+                if with_attr:
+                    attr_val += 1
+                    attributes["attr0"] = attr_val % 3
+                    attributes["attr%d" % (label_id + 1)] = attr_val % 2
+                for ann_id in range(count):
+                    append_bbox(
+                        annotations,
+                        label_id=label_id,
+                        ann_id=ann_id,
+                        attributes=attributes,
+                    )
+            item = DatasetItem(
+                id=str(img_id),
+                annotations=annotations,
+                subset=self._get_subset(img_id),
+                attributes={"id": img_id},
+            )
+            iterable.append(item)
+
+        dataset = Dataset.from_iterable(iterable, categories)
+        return dataset, totals
+
+    @staticmethod
+    def _get_append_bbox(dataset_type):
+        def append_bbox_coco(annotations, **kwargs):
+            annotations.append(
+                Bbox(
```

Comment:
Please fix the formatting.

Reply:
I've run black and check format with flake8 to follow pep8 format. 
If it is insufficient, can you give me more hints to resolve this?

Reply:
The code follows the rules in these specs, but it does not mean it is comfortable to read. Splitting a line on each parameter does not help with readability, in my opinion.

Reply:
I see. I've written bbox info in one line like below, but it seems that 'black' modifies these automatically.
```python
Bbox(
    1, 1, 2, 2, 
    label=...
)
```
I should remove 'black' from pre-commit hook or find another way to resolve this.
Anyway, I'll apply your comment. Thanks.